backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -279,13 +280,34 @@ def get_trees(self, hardware_id, origin, start_datetime, end_datetime):
 
         return trees
 
-    def get_selected_trees(self, trees, selected_trees):
-        def filter_fn(tree):
-            return selected_trees.get(tree.get('index')) == SELECTED_VALUE
+    def get_selected_trees(self, trees, selected_tree_param):
+        selected_trees = []
 
-        return list(filter(filter_fn, trees)) if selected_trees else trees
+        for tree in trees:
+            tree_idx = tree["index"]
+            selected_commit = selected_tree_param.get(tree_idx)
+
+            if not selected_tree_param or selected_commit == SELECTED_HEAD_TREE_VALUE:
+                selected_commit = tree['headGitCommitHash']
+            elif not selected_commit:
+                continue
+
+            selected_trees.append({
+                "tree_name": tree["treeName"],
+                "git_repository_branch": tree["gitRepositoryBranch"],
+                "git_repository_url": tree["gitRepositoryUrl"],
+                "index": tree["index"],
+                "git_commit_hash": selected_commit,
+            })
+
+        return selected_trees
+
+    def get_records(self, *, hardware_id, origin, start_datetime, end_datetime, trees):
+        commit_hashes = [tree["git_commit_hash"] for tree in trees]
+        qFilter = Q(build__checkout__start_time__gte=start_datetime)
+        qFilter &= Q(build__checkout__start_time__lte=end_datetime)
+        qFilter |= Q(build__checkout__git_commit_hash__in=commit_hashes)
```

Comment:
I don't know if this comment went trough because it disappeared, but I don't think we need to filter with timestamps here since we already have the commit hashes

Reply:
yes I notice that and I'm removing it right now